eng/TraversalSdk.AfterTargets.targets

```diff
@@ -9,10 +9,17 @@
     <VSTestDependsOn>$(VSTestDependsOn);ResolveP2PReferences</VSTestDependsOn>
     <PackDependsOn>$(PackDependsOn);ResolveP2PReferences</PackDependsOn>
     <PublishDependsOn>$(PublishDependsOn);ResolveP2PReferences</PublishDependsOn>
+    <GetTargetPathDependsOn>ResolveP2PReferences</GetTargetPathDependsOn>
+    <!-- Filter out ProjectReferences which aren't compatible with the project's TargetFramework. -->
+    <OmitIncompatibleProjectReferences>true</OmitIncompatibleProjectReferences>
```

Comment:
Why are we adding this?

Reply:
This already existed in the traversal projects: https://github.com/dotnet/runtime/blob/cdc7b457630332c46691ec32c6ba9c0010aa9f7f/src/libraries/src.proj#L9-L11.

The overall reason why this is necessary is already explained in https://github.com/dotnet/arcade/pull/8459. Incompatible target frameworks (based on the BuildTargetFramework property) were filtered out in outer builds. Now that many projects don't have outer builds anymore and use the `TargetFramework` property instead, filtering needs to happen at a different point. Note that this is already implemented in main, this change here just moves it into a shard place as multiple traversal projects depend on filtering.